src/delta_lake/delta_service.py

```diff
@@ -115,6 +118,74 @@ def _check_exists(database: str, table: str) -> bool:
     return True
 
 
+def _generate_cache_key(prefix: str, params: Dict[str, Any]) -> str:
+    """
+    Generate a cache key from a prefix and parameters.
+    """
+    # Convert parameters to a sorted JSON string to ensure consistency
+    param_str = json.dumps(params, sort_keys=True)
+    # Create a hash of the parameters to avoid very long keys
+    param_hash = hashlib.md5(param_str.encode()).hexdigest()
+    return f"{prefix}:{param_hash}"
+
+
+def _get_from_cache(
+    spark: SparkSession, cache_key: str
+) -> Optional[List[Dict[str, Any]]]:
+    """
+    Try to get data from Redis cache.
+    """
+    try:
+        cache_df = (
+            spark.read.format("org.apache.spark.sql.redis")
+            .option("table", cache_key)
+            .option("key.column", "id")
+            .load()
+        )
+
+        if not cache_df.rdd.isEmpty():
+            first_row = cache_df.first()
+            if first_row and "value" in first_row:
+                return json.loads(first_row["value"])
+
+    except Exception as e:
+        # Log the error but don't fail the operation if caching fails
+        logger.debug(f"Cache miss or error for key {cache_key}: {e}")
+
+    return None
+
+
+def _store_in_cache(
+    spark: SparkSession,
+    cache_key: str,
+    data: List[Dict[str, Any]],
+    ttl: int = CACHE_EXPIRY_SECONDS,
+) -> None:
+    """
+    Store data in Redis cache.
+    """
+    try:
+        json_data = json.dumps(data)
+
+        cache_data = [(1, json_data)]
+        cache_df = spark.createDataFrame(cache_data, ["id", "value"])
+
+        # Write to Redis
+        (
+            cache_df.write.format("org.apache.spark.sql.redis")
+            .option("table", cache_key)
+            .option("key.column", "id")
+            .option("ttl", ttl)
+            .mode("overwrite")
+            .save()
+        )
+
+        logger.debug(f"Cached data under key {cache_key} with TTL {ttl}s")
+    except Exception as e:
+        # Log the error but don't fail the operation if caching fails
+        logger.warning(f"Failed to cache data under key {cache_key}: {e}")
```

Comment:
Later we might want to have some way of monitoring this stuff - if we're getting unexpected caching failures and misses we'd probably want to know without having someone be lucky enough to see it when they're looking at the logs

Reply:
sure. I could be wrong - but I don't we are monitoring any of our services (KBase-wide) closely.

Reply:
We have health checks for all of them, which is all you need for most of the services.

Reply:
yup. That's also what I mean by `closely`. We only know it's up/down - we don't know/care anything about request rate, error rate, logs, etc.